pkg/recordbot/uploader_s3.go

```diff
@@ -0,0 +1,71 @@
+package recordbot
+
+import (
+	"errors"
+	"io"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+type S3Config struct {
+	Region string
+	Bucket string
+	ACL    string
+
+	// We use Role IAM object attached to EC2 instance.
+	RoleARN string
+}
+
+type s3Uploader struct {
+	config S3Config
+	client s3iface.S3API
+}
+
+var ErrEmptyS3BucketName = errors.New("empty S3 bucket name")
+var ErrEmptyS3ACL = errors.New("empty S3 ACL")
+
+func NewS3Uploader(config S3Config) (Uploader, error) {
```

Comment:
I'm conflicted if this belongs in livekit-sdk. It pulls in aws-sdk-go and is quite platform specific. wdyt about having users handle this on their own? by passing in an appropriate uploader interface.

Reply:
Yeah I thought so too. The idea was to provide a plug-and-play functionality to upload to S3. But I realised this is better left for the API consumer as this introduces unnecessary dependencies. I’ll delete the S3 code and tidy up the dependencies